server/webgis/auth/decorators.py

```diff
@@ -0,0 +1,39 @@
+import base64
+from functools import wraps
+
+from django.http import HttpResponse
+from django.views.decorators.vary import vary_on_headers
+# from django.utils.decorators import available_attrs
+from django.contrib.auth.models import AnonymousUser
+
+from . import basic_auth
+
+# def basic_authentication(realm='API'):
```

Comment:
So, this is no longer needed?

Reply:
Basic authentication is currently used in /api/map/ows/ endpoint:
https://github.com/gislab-npo/gisquick/blob/api-redesign/server/webgis/map/views.py#L48

I think we cannot use decorator for this kind of authentication, because it depends on project settings, and we don't have access to this kind of information in decorator. For public projects it shouldn't be checked, otherwise browser will ask for password, if decorator return 401 response with WWW-Authenticate header. That's why this functionality was moved into ows view, after project permissions check. It's not definitely impossible to use decorator in the future, but we should have unified way of passing project info in API requests, but currently with single endpoint with basic auth support, it's probably fine to handle this functionality inside view.